odoo/addons/base/models/ir_ui_view.py

```diff
@@ -62,7 +62,7 @@ def att_names(name):
 def transfer_field_to_modifiers(field, modifiers, view_editable=True):
     default_values = {}
     state_exceptions = {}
-    attributes = ('invisible', 'readonly', 'required') if view_editable else ('invisible',)
+    attributes = ('invisible', 'readonly', 'required') if view_editable else ('invisible', 'readonly')
```

Comment:
cc: @beledouxdenis

Reply:
Indeed, this goes against odoo/odoo#101924.

It's a bummer to have to suddenly set the `readonly` attribute for all readonly fields in all view types (therefore including graph, pivot, ...) just because you need it for the color widget in tree views *-).

In addition, the web client has already the information the field is readonly without having to set it in the view architecture itself.

Instead, you should alter the Javascript code of the widget color to get the readonly attribute from the fields description if it's not set in the view architecture.

Reply:
If js team is ok with the updated solution client side, ok for me.
(The base problem comes from making widgets editable in non editable views, though it seems we really want to keep supporting that: a818a9b73265949a5d1d7f13b82538137e5a5ad3)

Reply:
Mouarf I'm not confident with this. The client code has been written with the assumption that the readonly information was to read from modifiers, and that the field one could be ignored. That bothers me that we need to read the information from somewhere, and sometimes from somewhere else. This doesn't look like a well designed API and is error prone. For instance, there is at least [another place](https://github.com/odoo/odoo/blob/16.0/addons/web/static/src/views/fields/field.js#L58) where the same fallback should be done (I guess).

@beledouxdenis I'm not sure to know the server-side logic about setting the readonly modifier or not, but I assume there is some sort of specific conditions for list and kanban views to determine whether or not the information is relevant. If so, don't you think this went too far? There's a trade-of between code complexity and efficiency. I'm wondering whether it would be better to simply always send this information for form, list and kanban views. And if we want to save some bytes, we could remove the readonly information from the field description as it is often redundant with the modifier.